.github/workflows/publish.yml

```diff
@@ -34,13 +36,16 @@ jobs:
 
       - name: ❓ Check if a new version needs to be published
         id: publish-check
-        continue-on-error: true
-        run: yarn publish:check
+        run: echo "SHOULD_PUBLISH=$(yarn publish:check)" >> "$GITHUB_OUTPUT"
```

Comment:
Previous version was using the exit code of `yarn publish:check` to define if a real publishing was needed or not. This worked, but this was swallowing any legitimate error in the `yarn publish:check` script, and considered them as normal behavior.

I'm now using GitHub Actions output (see: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) to pass values from step to step.

Here I define an output called `SHOULD_PUBLISH`, with a value returned by `yarn publish:check` (that will be either `yes` or `no`), and I'll use that later to conditionally execute some steps.